Sources/Sentry/SentryNetworkTrackingIntegration.m

```diff
@@ -16,12 +17,14 @@ - (void)installWithOptions:(SentryOptions *)options
     self.options = options;
     if (options.enableAutoHttpRequestTracking) {
         [SentryNetworkSwizzling start];
+        [NSURLProtocol registerClass:[SentryHttpInterceptor class]];
     }
 }
 
 - (void)uninstall
 {
     [SentryNetworkSwizzling stop];
+    [NSURLProtocol unregisterClass:[SentryHttpInterceptor class]];
```

Comment:
`m`: Can we add tests to validate that we register and unregister to `NSURLProtocol`?

Reply:
There's no API to check if a protocol was registered.

Reply:
We could check if a registered interceptor still gets called, after `uninstall`. This way we know we called `[NSURLProtocol unregisterClass]`